agent/bench-scripts/pbench-user-benchmark

```diff
@@ -120,14 +125,28 @@ export benchmark_results_dir=$benchmark_results_dir
 export benchmark config
 pbench-metadata-log --group=$tool_group --dir=$benchmark_run_dir beg
 
+# Add data to metadata.log
+function record_iteration {
```

Comment:
There is only ever one caller, but it's messy enough in general that encapsulating it is a good idea. Every bench-script has its own version of this, so we are trying to be consistent.